binderhub/builder.py

```diff
@@ -258,7 +258,10 @@ def escape(s):
         self.ref_url = await provider.get_resolved_ref_url()
         resolved_spec = await provider.get_resolved_spec()
 
-        self.binder_launch_host = self.settings['badge_base_url'] or '{proto}://{host}{base_url}'.format(
+        badge_base_url = self.settings['badge_base_url']
+        if callable(badge_base_url):
+            badge_base_url = badge_base_url(self)
```

Comment:
I think it would be better to put these 3 lines into a method/property `badge_base_url` of `BaseHandler` and use it

Reply:
Done!